lib/postcode_validation/gateway/pca_address_list.rb

```diff
@@ -9,33 +9,44 @@ class PCARequestError < PostcodeValidation::Error::RequestError; end
       base_uri 'https://services.postcodeanywhere.co.uk'
 
       def query(search_term:, country:)
-        response = address_list_for_postcode(country, search_term)
-
-        response.map do |row|
+        address_list_for_postcode(country, search_term).map do |row|
           raise PCARequestError, error_message(row) if row.key?('Error')
 
-          PostcodeValidation::Domain::Address.new(street_address: row['StreetAddress'],
-                                                  place: row['Place'])
-        end
+          address_payload(row, country, search_term)
+        end.compact.flatten
       end
 
       private
 
+      def address_payload(row, country, search_term)
+        if row['Type'] == 'Address'
+          PostcodeValidation::Domain::Address.new(street_address: formatted_response(row))
+        elsif row['Type'] == 'Postcode'
+          PostcodeValidation::Gateway::PCAPostcodeToAddresses.new(
+            search_term: search_term,
+            country: country
+          ).execute
+        end
```

Comment:
From what I can tell, yeah.  If not a bunch of nils will be mapped into an array, but then compacted out.  So it will end up being an empty array and still safe.